packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-link/gux-rich-text-editor-action-link.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+  Component,
+  Prop,
+  State,
+  Element,
+  EventEmitter,
+  Event,
+  Listen,
+  h,
+  Host
+} from '@stencil/core';
+import { OnClickOutside } from '@utils/decorator/on-click-outside';
+import { trackComponent } from '@utils/tracking/usage';
+import { buildI18nForComponent, GetI18nValue } from 'i18n';
+import translationResources from '../i18n/en.json';
+import { afterNextRender } from '@utils/dom/after-next-render';
+
+@Component({
+  tag: 'gux-rich-text-editor-action-link',
+  styleUrl: 'gux-rich-text-editor-action-link.scss',
+  shadow: true
+})
+export class GuxRichTextEditorActionLink {
+  private i18n: GetI18nValue;
+  actionButton: HTMLElement;
+  linkAddressInputElement: HTMLInputElement;
+  textToDisplayInputElement: HTMLInputElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  @Prop()
+  disabled: boolean = false;
```

Comment:
A question from UX was should we disable this `link` action until a piece of text is highlighted ? I